internal/controller/article.go

```diff
@@ -0,0 +1,77 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"news/internal/entity"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type articleInf interface {
+	GetArticles(context.Context) ([]*entity.NewsletterNewsItem, error)
+	GetArticle(context.Context, int64) (*entity.NewsArticle, error)
+}
+
+type ArticleHandler struct {
+	article articleInf
+	ctx     context.Context
```

Comment:
why do you inject context into a handler struct? 🤔 Isn't it supposed to be different for every request?

Reply:
Fixed